plugins/depthcamera/DepthCameraDriver.cpp

```diff
@@ -0,0 +1,273 @@
+{
+    return getRgbFOV(horizontalFov, verticalFov);
+}
+
+bool DepthCameraDriver::setDepthFOV(double horizontalFov, double verticalFov)
+{
+    yCError(GZDEPTH) << "setDepthFOV: impossible to set FOV";
+    return false;
+}
+
+
+bool DepthCameraDriver::getDepthIntrinsicParam(yarp::os::Property& intrinsic)
+{
+    return getRgbIntrinsicParam(intrinsic);
+}
+
+double DepthCameraDriver::getDepthAccuracy()
+{
+    return 0.00001;
+}
+
+bool DepthCameraDriver::setDepthAccuracy(double accuracy)
+{
+    yCError(GZDEPTH)  << "setDepthAccuracy: impossible to set accuracy";
+    return false;
+}
+
+bool DepthCameraDriver::getDepthClipPlanes(double& nearPlane, double& farPlane)
+{
+    if(!m_sensorData)
+    {
+        yCError(GZDEPTH) << "getDepthClipPlanes: sensor data not available!";
+        return false;
+    }
+    nearPlane = m_sensorData->nearPlane;
+    farPlane  = m_sensorData->farPlane;
+    return true;
+}
+
+bool DepthCameraDriver::setDepthClipPlanes(double nearPlane, double farPlane)
+{
+    yCError(GZDEPTH)  << "setDepthClipPlanes: impossible to set clip planes";
+    return false;
+}
+
+bool DepthCameraDriver::getDepthMirroring(bool& mirror)
+{
+    mirror = false;
+    return true;
+}
+
+bool DepthCameraDriver::setDepthMirroring(bool mirror)
+{
+    yCError(GZDEPTH)  << "setDepthMirroring: impossible to set mirroring";
+    return false;
+}
+
+bool DepthCameraDriver::getDepthImage(yarp::sig::ImageOf<yarp::sig::PixelFloat>& depthImage, yarp::os::Stamp* timeStamp)
+{
+    if(!m_sensorData)
+    {
+        yCError(GZDEPTH) << "gazeboDepthCameraDriver: sensor data not available!";
+        return false;
+    }
+    if(!timeStamp)
+    {
+        yCError(GZDEPTH)  << "gazeboDepthCameraDriver: timestamp pointer invalid!";
+        return false;
+    }
+    std::lock_guard<std::mutex> lock(m_sensorData->m_mutex);
+    depthImage.resize(m_sensorData->m_width, m_sensorData->m_height);
+    if(!m_depthQuantizationEnabled) {
+        memcpy(depthImage.getRawImage(), m_sensorData->depthCameraMsg.data().c_str(), m_sensorData->depthCameraMsg.ByteSizeLong());
+    }
+    else {
+        float nearPlane = (float) m_sensorData->nearPlane;
+        float farPlane = (float) m_sensorData->farPlane;
+        int intTemp;
+        float value;
+        float powCoeff = pow(10.0f, (float) m_depthDecimalNum);
+
+        auto pxPtr = reinterpret_cast<float*>(depthImage.getRawImage());
+        for(int i=0; i< m_sensorData->depthCameraMsg.data().size(); i++){
+            value = m_sensorData->depthCameraMsg.data()[i];
+
+            intTemp = (int) (value * powCoeff);
+            value = ((float) intTemp) / powCoeff;
+
+            if (value < nearPlane) { value = nearPlane; }
+            if (value > farPlane) { value = farPlane; }
+
+            *pxPtr = value;
+            pxPtr++;
+        }
+    }
+    timeStamp->update(m_sensorData->simTime);
+    //TODO vertical and horizontal flip, display timestamp and time box
```

Comment:
Can you open an issue to track the TODO or remove it? TODO in the code are a bit difficult to track and manage.

Reply:
Yes actually this functionality is not present in the gazebo-yarp-plugins depth camera, I would remove it and if someone needs it I will add it

Reply:
Sure!